lib/puma/cluster.rb

```diff
@@ -474,6 +491,29 @@ def run
 
       @launcher.events.fire_on_booted!
 
+      Thread.new do
+        Puma.set_thread_name "acceptor"
+
+        # Chooses a load-balancing algorithm from a set of predefined symbol constants or a custom Proc.
+        algorithms = {
+          least_busy: lambda {|workers| workers.min_by {|w| [-w.capacity, w.last_request]}},
+          round_robin: lambda {|workers| workers.min_by(&:last_request)},
+          pile_up: lambda {|workers| workers.select {|w| w.capacity > 0}.min_by {|w| w.index} || workers.min_by(&:last_request)},
+          random: lambda {|workers| workers.sample}
+        }.freeze
+        routing = @options[:load_balancing]
+        debug "Using load balancing algorithm: #{routing}"
+        routing = algorithms[routing] unless routing.is_a?(Proc)
+        raise "Invalid value for load_balancing - #{@options[:load_balancing]}" unless routing.is_a?(Proc)
```

Comment:
We should probably be checking this in the config DSL rather than at this level.